### PR TITLE
Upgrade Test Improvements

### PR DESCRIPTION
**Pull request by chatton**
## Description

**Review after #4649**

This PR makes a few small improvements to the upgrade tests.

- Fixes the v6 - v7 upgrade test (it was failing only with hermes, a restart is required - the issue is linked) thank you @colin-axner for the help debugging this one!
- Adds extra upgrade tests to the upgrade workflow, we will now run all the upgrade tests with both hermes and rly.
- Adds additional IBC transfers post upgrade (verify relayers are still working)

closes: #4216

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]
Awesome! Will we be able to remove this conditional on restarting hermes eventually? Do we need an issue of our own to track it? 

LGTM 🚀

**Comment by chatton** [comment]
> Awesome! Will we be able to remove this conditional on restarting hermes eventually? Do we need an issue of our own to track it?
> 
> LGTM 🚀

At the moment there's no tracker on our end, there is [this issue](https://github.com/informalsystems/hermes/issues/3579#issuecomment-1719088315). But also I don't think it's a huge priority to refactor the upgrade test to not have the restart.

File: e2e/tests/transfer/base_test.go [hunk 1]
```diff
@@ -103,6 +103,10 @@ func (s *TransferTestSuite) TestMsgTransfer_Succeeds_Nonincentivized() {
 
 		expected := testvalues.IBCTransferAmount
 		s.Require().Equal(expected, actualBalance.Int64())
+
+		if testvalues.DenomMetadataFeatureReleases.IsSupported(chainBVersion) {
```
Comment (charleenfei) [inline e2e/tests/transfer/base_test.go:107]: 🚀

**Review by charleenfei — APPROVED** [review]
nice work @chatton!

**Pull request merged**